example/android/app/build.gradle

```diff
@@ -64,4 +64,5 @@ dependencies {
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'androidx.test:runner:1.1.1'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
+    implementation 'com.android.installreferrer:installreferrer:2.2'
```

Comment:
Good question. I also thought that this dependency should be only in module but unfortunately right now build fails without this dependency.

Reply:
I found the issue... installreferrer is listed as `implemention 'com...` dependency in the libs build.gradle. While this includes the dependency in hosting projects (like the example app), the example app can however not use APIs provided by installreferrer itself.

This is from the Android documentaion:

> Gradle adds the dependency to the compile classpath and packages the dependency to the build output. However, when your module configures an implementation dependency, it's letting Gradle know that you do not want the module to leak the dependency to other modules at compile time. That is, the dependency is available to other modules only at runtime.

Flutter injects some code like
```
flutterEngine.getPlugins().add(new de.lschmierer.android_play_install_referrer.AndroidPlayInstallReferrerPlugin())
```
into the example project. SInce you added the `InstallReferrerStateListener ` interface to `AndroidPlayInstallReferrerPlugin`, installreferrer API is now part of the 
android_play_install_referrer API.

The easiest solution would be to change
``implementation 'com.android.installreferrer:installreferrer:2.2'` to `api 'com.android.installreferrer:installreferrer:2.2'` in https://github.com/lschmierer/android_play_install_referrer/blob/master/android/build.gradle

However, I would prefer to not pollute the the hosting packages scope by not exposing installreferrer APIs.
I.e. to not have `AndroidPlayInstallReferrerPlugin` implement `InstallReferrerStateListener `.